src/videojs-contrib-hls.js

```diff
@@ -106,11 +86,17 @@ const getHlsPlaylistIndex = function(media, playlists) {
  */
 const handleHlsMediaChange = function(qualityLevels, playlistLoader) {
   let newPlaylist = playlistLoader.media();
-  let selectedIndex = getHlsPlaylistIndex(newPlaylist, playlistLoader.master.playlists);
+  let i;
 
-  qualityLevels.selectedIndex_ = selectedIndex;
+  for (i = qualityLevels.length - 1; i >= 0; i--) {
```

Comment:
Was just trying to avoid an extra check. If no quality level is found with the correct id, i will become -1 at the end of the loop, which is what we want to set selectedIndex if there isnt a selected QualityLevel, but along with your comment above, it definitely makes things less clear and its nothing extra to restructure this in a more clear way